Firestore/Source/API/FIRQuery.m

```diff
@@ -107,7 +107,7 @@ - (instancetype)initWithQuery:(FSTQuery *)query firestore:(FIRFirestore *)firest
 
 - (BOOL)isEqual:(nullable id)other {
   if (other == self) return YES;
-  if (!other || ![[other class] isEqual:[self class]]) return NO;
+  if (![[other class] isEqual:[self class]]) return NO;
```

Comment:
If you haven't already, can you just double-check that this still works?  I'm a tiny bit nervous because AppCode auto-generates these methods with the "!other" check, and I don't know why.

Reply:
Still works, see logic given below:

The only difference happens when `other` is `nil`. That case one return `NO` immediately while the other will check `![[other class] isEqual:[self class]]`, which is equivalent to `![nil isEqual:[self class]]`, which is equivalent to `!nil`, which makes the method return `NO` as well. So `!other` check is redundant.

I have no particular preference on one over the other. So I can change it back if you think with `!other` check is easier to understand.